app/src/main/java/com/smelnykov/homework_1/MyProfileActivity/MyProfileActivity.kt

```diff
@@ -32,4 +44,34 @@ class MyProfileActivity : AppCompatActivity() {
         val name = binding.appCompatTextName
         name.text = parser.parseNameSurname(email)
     }
+
+    /**
+     * Method sets up the click listeners for log out button.
+     */
+    private fun setClickListeners() {
+
+        val buttonLogOut = binding.appCompatButtonLogOut
+
+        buttonLogOut.setOnClickListener {
+            clearData()
+            logOutRedirection()
+        }
+    }
+
+    /**
+     * Method clears users data in sharedPreference.
+     * */
+    private fun clearData() {
```

Comment:
mentioned before, best have work with shared preferences in a separate util class or object

Reply:
Thank you! Made changes. Commit 9e0bcee7